go/client/cmd_chat_inbox.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package client
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	"github.com/keybase/cli"
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+	"github.com/keybase/gregor/protocol/chat1"
+	"github.com/keybase/gregor/protocol/gregor1"
+)
+
+type uidUsernameMapper map[keybase1.UID]libkb.NormalizedUsername
+
+func (m uidUsernameMapper) getUsername(g *libkb.GlobalContext, uid keybase1.UID) (username libkb.NormalizedUsername, err error) {
+	if m == nil {
+		m = make(uidUsernameMapper)
+	}
+
+	if username, ok := m[uid]; ok {
+		return username, nil
+	}
+
+	loadUserArg := libkb.NewLoadUserArg(g)
+	loadUserArg.UID = uid
+
+	var ret *libkb.User
+	if ret, err = libkb.LoadUser(loadUserArg); err != nil {
```

Comment:
I don't think we ever call this in the `client` directory.  We instead call the RPC `LoadUserByName`